lib/puppet/provider/sshkey/parsed.rb

```diff
@@ -36,5 +27,25 @@
   def self.default_mode
     0644
   end
+
+  def self.default_target
+    case Facter.value(:operatingsystem)
+    when "Darwin"
+      # Versions 10.11 and up use /etc/ssh/ssh_known_hosts
+      version = Facter.value(:macosx_productversion_major)
+      if version
+        major, minor = version.split('.')
+        if major.to_i > 10 || (major.to_i == 10 && minor.to_i >= 11)
```

Comment:
You might be able to simplify this logic using `Puppet::Util::Package.versioncmp(version, '10.11') >= 0`